src/main/java/qna/service/QnAService.java

```diff
@@ -35,24 +35,15 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
+        question.checkLoginUser(loginUser);
+
+        Answers answers = Answers.from(question.getAnswers());
+        answers.checkOwners(loginUser);
+
+        DeleteHistories deleteHistories = new DeleteHistories();
+
+        question.setDeletedTrue();
```

Comment:
question 삭제시 DeleteHistories를 반환하도록 하면 어떨까요 :) ?
```suggestion
        DeleteHistories deleteHistories = question.delete();
```